gtsam/slam/OrientedPlane3Factor.h

```diff
@@ -48,10 +49,27 @@ class OrientedPlane3Factor: public NoiseModelFactor2<Pose3, OrientedPlane3> {
   virtual Vector evaluateError(const Pose3& pose, const OrientedPlane3& plane,
       boost::optional<Matrix&> H1 = boost::none, boost::optional<Matrix&> H2 =
           boost::none) const {
-    OrientedPlane3 predicted_plane = OrientedPlane3::Transform(plane, pose, H1,
-        H2);
     Vector err(3);
-    err << predicted_plane.error(measured_p_);
+
+    if (H1 || H2) {
+      Matrix H1_1, H2_1;
```

Comment:
So, the static method is deprecated (wrong Jacobian order), so please use the method. Give these better names and make them fixed-size matrices, like

```
Matrix33 predicted_H_plane;
Matrix36 predicted_H_pose;
auto predicted = plane.transform(pose, predicted_H_plane, predicted_H_pose);
```